algorithms/src/snark/marlin/data_structures/proof.rs

```diff
@@ -131,99 +147,128 @@ impl<F: PrimeField> Evaluations<F> {
         mut writer: W,
         compress: Compress,
     ) -> Result<(), snarkvm_utilities::SerializationError> {
-        for z_b_eval in &self.z_b_evals {
-            CanonicalSerialize::serialize_with_mode(z_b_eval, &mut writer, compress)?;
-        }
+        CanonicalSerialize::serialize_with_mode(&self.z_b_evals, &mut writer, compress)?;
         CanonicalSerialize::serialize_with_mode(&self.g_1_eval, &mut writer, compress)?;
-        CanonicalSerialize::serialize_with_mode(&self.g_a_eval, &mut writer, compress)?;
-        CanonicalSerialize::serialize_with_mode(&self.g_b_eval, &mut writer, compress)?;
-        CanonicalSerialize::serialize_with_mode(&self.g_c_eval, &mut writer, compress)?;
+        CanonicalSerialize::serialize_with_mode(&self.g_a_evals, &mut writer, compress)?;
+        CanonicalSerialize::serialize_with_mode(&self.g_b_evals, &mut writer, compress)?;
+        CanonicalSerialize::serialize_with_mode(&self.g_c_evals, &mut writer, compress)?;
```

Comment:
This will serialize the lengths of each vector also, which will duplicate what's already in the serialization of `Proof`.

Reply:
I think we should add a helper function `serialize_vec_without_len(s: Vec<impl CanonicalSerialize>, writer: mut impl Write)` and a `BTreeMap` equivalent. Similarly for `vec_serialize_size_without_len` and `deserialize_vec_without_len`.

Reply:
Something like this? Sharing it here for comments as I won't finish a full implementation tonight still.

```
    fn serialize_vec_without_len(
        src: &Vec<impl CanonicalSerialize>, 
        mut writer: impl Write, 
        compress: Compress
    ) -> Result<(), snarkvm_utilities::SerializationError> {
        for elem in src {
            CanonicalSerialize::serialize_with_mode(elem, &mut writer, compress)?;
        }
        Ok(())
    }

    fn serialized_vec_size_without_len(
        src: &Vec<impl CanonicalSerialize>, 
        compress: Compress
    ) -> usize {
        let mut size = 0;
        for elem in src {
            size += CanonicalSerialize::serialized_size(elem, compress);
        }
        size
    }

    fn deserialize_vec_without_len(
        mut reader: impl snarkvm_utilities::Read,
        compress: Compress,
        validate: Validate,
        len: usize,
    ) -> Result<(), snarkvm_utilities::SerializationError> {
        let mut res = (0..len).map(|_| {
            CanonicalDeserialize::deserialize_with_mode(&mut reader, compress, validate)
        }).collect::<Result<Vec<F>, _>>()?;
        Ok(())
    }
```